src/components/Icon/Icon.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './Icon.scss';
+import { CONTEXTS, ICONS } from '../../constants';
+
+const { block, elem } = bem({
+    name: 'Icon',
+    classnames: styles,
+    propsToMods: ['context', 'margin']
+});
+
+const Icon = props => {
+    const { context, name, size, title, ...rest } = props;
+
+    const currentIcon = ICONS[name];
```

Comment:
I'd suggest to implement this bit differently (see below why):

### 1. To create a base IconBase
Component that is basically just creates an <svg> element with all the boring default <svg> attrs like:
```js
<svg
  xmlns="http://www.w3.org/2000/svg"
  stroke="currentColor"
  fill="currentColor"
  strokeWidth="0"
  ... // all common svg stuff
>
  {children}
</svg>
```

### 2. To create a IconLogo as a common wrapper for all logo icons

```js
import IconBase from './IconBase';

...

return (
    <div {...rest} {...block(props)}>
        <IconBase
            {...elem('svg', props)}>
            viewBox={currentIcon.viewBox}
            ... // all common logo icon stuff
        >
            {!!title && <title>{title}</title>}
            {children}
        </svg>
    </div>
);
```

### 3. And for each product logo create actual component

**./src/components/logos/ExtractLogo/ExtractLogo.js**
```js
import IconLogo from './IconLogo';

return (
  <IconLogo viewBox="0 0 77.36 93.32">
    <path d="M5.17,87.36l-.46,4c0,1,2.14,1.91,4.71,1.91H67.93c2.58,0,4.71-.9,4.71-1.91l-.45-4Z" />
    <path d="M2.81,78.6l-.45,3.71c0,1.12,2.25,2,5.05,2H70c2.81,0,5-.9,5-2l-.56-3.71Z'" />
    <path d="M75.35,57.15C75.35,55.91,73,55,70,55H55v3.81a3.71,3.71,0,1,1-7.42,0V55H28.75v3.81a3.71,3.71,0,1,1-7.42,0V55H7.3c-2.91,0-5.39,1-5.39,2.13L0,73.55c0,1.24,2.36,2.13,5.39,2.13H72c2.91,0,5.38-1,5.38-2.13Z" />
     ...
  </IconLogo>
);
```

## Why

1. Keeping all the icons as constants in  **src/constants/icons.js** will force us (users of oneUI) to load all icons in a bundle when we need only one of them.
2. `<svg>` element can contain not only `<path>` but a lot of other tags like `<circle>` for instance.
3. `<IconBase/>` can be nicely reused later anytime we want to use svg

Reply:
Ok, I think that all makes sense. But do you really think we also need `IconLogo`? It seems sort of useless to me. The outer div as well as the optional title can also be part of `IconBase`, right?

Then, in terms of naming, I would maybe go with `Icon` and `IconProduct`

Reply:
Having `IconBase` may be indeed too much for just `<svg/>`.

In terms of naming I think make sense to name base icon as `ProductIcon` or `ProductLogo` or just `Logo`. Point here is that just `Icon` is too abstract and in this case we deal only with our logos.

So I'd go with something like `Logo` and then bunch of `SearchLogo`, `ExtractorLogo`, `JobfeedLogo` etc...

Reply:
...apparently my comment was deleted because I didn't finish the 'review' 🙄 

Anyway, so I would then suggest to have `IconBase` which is a combination of what you first suggested to be `IconBase` and `IconLogo` (don't see a reason for those to be two different components), and then a component for each distinct icon: `IconSearch`, `IconJobfeed` etc.

The tree structure could be something like this:

```
└── components
    └── Icon
        ├── IconBase
        │   ├── IconBase.js
        │   ├── IconBase.scss
        │   └── tests
        ├── IconSearch
        │   ├── IconSearch.js
        │   ├── IconSearch.scss
        │   └── tests
        └── IconJobfeed
            ├── IconJobfeed.js
            ├── IconJobfeed.scss
            └── tests
```
Makes sense?

Reply:
Yes, looks good!
Except one thing, as I said above, `Icon` is a too generic name as for me.
Put yourself in a shoes of a new developer who come to our project and sees this:
```
├── components
│   ├── Alert
│   ├── Button
│   ├── ...
│   ├── Icon
│   ├── Link
```

No way that he/she will guess that it is about product logos.

Reply:
But it does not necessarily have to be only product icons, right? Can be any custom icon of ours, who knows.. Although for UI purposes we will probably be using another package like FA5 Pro.

Reply:
Exactly, something like https://github.com/react-icons/react-icons would make more sense as a icon pack instead of having it in the OneUI lib.